synctl

```diff
@@ -292,11 +319,13 @@ def main():
             write("All processes exited; now restarting...")
 
     if action == "start" or action == "restart":
+        error = False
         if start_stop_synapse:
             # Check if synapse is already running
             if os.path.exists(pidfile) and pid_running(int(open(pidfile).read())):
                 abort("synapse.app.homeserver already running")
-            start(configfile, bool(options.daemonize))
+
+            error |= start(configfile, bool(options.daemonize))
```

Comment:
think this is still backwards.

I think it would be much clearer if you spelt it out:

```
success = start(...)
if not success:
    error = True
```